nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -511,46 +511,29 @@ Costmap2DROS::getRobotPose(geometry_msgs::msg::PoseStamped & global_pose)
   tf2::toMsg(tf2::Transform::getIdentity(), robot_pose.pose);
 
   robot_pose.header.frame_id = robot_base_frame_;
-  robot_pose.header.stamp = rclcpp::Time();
```

Comment:
Yes and no, I left that as it was with fix ups because there's lot of implicit things Costmap2D already has access to (frames, timeouts, logger, a buffer). I'd thought about it but decided against it, you think we should? I think it would make inline code uglier in costmap 2D because of how much we have to specify for that particular usecase of the default parameters

Reply:
As a compromise, perhaps this function could call the `nav2:util::getCurrentPose` with the proper defaults.

Reply:
Fair enough. I have alot of blood, sweat, and tears in `costmap_2d` so whenever I gut it breaks my heart a little 💔 

but you're right